### ISSUE TITLE
webpack build hangs on Windows 10

### ISSUE THREAD

**Issue by GalmWing** [webpack-contrib/thread-loader#37]
Whenever I add thread-loader webpack hangs when building.

>      {
>         test: /\.jsx?$/,
>         use: [
>           { loader: "thread-loader" }, 
>           { loader: "babel-loader" }
>         ],
>         exclude: /node_modules/
>       }

I left it overnight just to test and webpack never finished, it should usually come back in less than 1 minute. I've also tried adjusting the number of workers and the poolTimeout to no avail. Removing thread-loader does fix the issue.

I'd like to add that my incremental build (aka webpack --watch) is faster with thread-loader and can keep building without issues.

I've also seen that HappyPack has some bugs either open or closed that show the same behavior. And my project also hangs if I replace thread-loader with happyPack. Apparently they solved their issues by fixing some fork() code. Though, it seems it didn't work for me. I'm wondering if there something on Windows that is preventing the process from returning.

**Comment by mistic** [comment]
@GalmWing can you provide mor info about the context where your problem is happening like: windows version, shell you're using on windows, full webpack config, how are you calling the webpack ... I've used with success the thread-loader over windows 10 and the default windows shell.

**Comment by evilebottnawi** [comment]
@a-aslan try https://github.com/webpack-contrib/thread-loader/pull/52

**Comment by a-aslan** [comment]
I'll wait until you guys merge and release this PR

**Comment by evilebottnawi** [comment]
@a-aslan Do you use `WSL`?

**Comment by jantimon** [comment]
@a-aslan does it work with thread-loader 1.2 ?

**Comment by mistic** [comment]
@a-aslan are u sure u are running `thread-loader@2.0.1`?

**Comment by a-aslan** [comment]
@mistic yes, I was running `thread-loader 2.0.1`, I'm gonna try it with `1.2.0` now.

**Comment by jantimon** [comment]
Had the same problem on mac

**Comment by mistic** [comment]
@jantimon @a-aslan can you provide some basic setup repo so I can test it?

**Comment by a-aslan** [comment]
In my prove gifv, I forgot to save the webpack file after i uncommented the thread-loader, but I just retried it, and I can confirm it still works after saving.

**Comment by mistic** [comment]
@jantimon @a-aslan it should be fixed on https://github.com/webpack-contrib/thread-loader/pull/53

**Comment by a-aslan** [comment]
Thanks, forgot to report feedback that in `v2.1.1` the problem is indeed resolved <3

Thanks to `thread-loader` our webpack build time has improved from **2:30** minutes to **45 seconds**!!!
